Corona-Warn-App/src/main/java/de/rki/coronawarnapp/presencetracing/warning/storage/TraceWarningRepository.kt

```diff
@@ -0,0 +1,182 @@
+package de.rki.coronawarnapp.presencetracing.warning.storage
+
+import android.content.Context
+import de.rki.coronawarnapp.diagnosiskeys.server.LocationCode
+import de.rki.coronawarnapp.presencetracing.warning.WarningPackageId
+import de.rki.coronawarnapp.util.HourInterval
+import de.rki.coronawarnapp.util.TimeStamper
+import de.rki.coronawarnapp.util.di.AppContext
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.map
+import timber.log.Timber
+import java.io.File
+import java.io.IOException
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
```

Comment:
Reusable should be sufficient

Reply:
You are right that this only accessed within `PresenceTracingWarningTask`. I'd still prefer to keep the guarantees provided by @Singleton against double instantiation here, I would in later PRs see if we can display the metadata that was processed to a test fragment.